routing.py

```diff
@@ -0,0 +1,223 @@
+    """
+    Check if a string is one of the available versions
+    """
+
+    versions = get_versions()
+
+    if versions:
+        return version in versions
+
+
+class YamlRegexMap:
+    def __init__(self, filepath):
+        """
+        Given the path to a YAML file of RegEx mappings like:
+
+            hello/(?P<person>.*)?: "/say-hello?name={person}"
+            google/(?P<search>.*)?: "https://google.com/?q={search}"
+
+        Return a list of compiled Regex matches and destination strings:
+
+            [
+                (<regex>, "/say-hello?name={person}"),
+                (<regex>, "https://google.com/?q={search}"),
+            ]
+        """
+
+        self.matches = []
+
+        if os.path.isfile(filepath):
+            with open(filepath) as redirects_file:
+                lines = yaml.load(redirects_file)
+
+                if lines:
+                    for url_match, target_url in lines.items():
+                        # Make sure all matches start with slash
+                        if url_match[0] != '/':
+                            url_match = '/' + url_match
+
+                        self.matches.append(
+                            (re.compile(url_match), target_url)
+                        )
+
+    def get_target(self, url_path):
+        for (match, target) in self.matches:
+            result = match.fullmatch(url_path)
+
+            if result:
+                parts = {}
+                for name, value in result.groupdict().items():
+                    parts[name] = value or ''
+                return target.format(**parts)
+
+
+class TemplateFinder:
+    """
+    Provides functions for matching URL paths to templates
+    """
+
+    def __init__(self, templates_dir):
+        """
+        Initialise the class with the path to the templates we should use
+        """
+
+        self.templates_dir = templates_dir
+
+    def get_languages(self, preferred_order=[]):
+        """
+        Find the available languages that exist as template folders
+        """
+
+        available_languages = []
+        top_folders = os.listdir(self.templates_dir)
+
+        for folder in top_folders:
+            if is_language(folder):
+                available_languages.append(folder)
+
+        weighted_languages = []
+
+        if available_languages and preferred_order:
+            for item in available_languages:
+                try:
+                    weight = preferred_order.index(item)
+                except ValueError:
+                    weight = available_languages.index(item) + len(
+                        preferred_order
+                    )
+
+                weighted_languages.append(
+                    (weight, item)
+                )
+
+            weighted_languages = sorted(weighted_languages)
+
+            available_languages = [item[1] for item in weighted_languages]
+
+        return available_languages
+
+    def find_alternate_path(self, request_path, languages, versions):
```

Comment:
This is a really long function, is it possible to split it in small functions?
For example a function per alternative path?